libs/execution_context/ebpf_core.c

```diff
@@ -2384,5 +2382,22 @@ ebpf_core_invoke_protocol_handler(
 bool
 ebpf_core_cancel_protocol_handler(_Inout_ void* async_context)
 {
-    return ebpf_async_cancel(async_context);
+    ebpf_epoch_enter();
+    bool return_value = ebpf_async_cancel(async_context);
+    ebpf_epoch_exit();
+    return return_value;
+}
+
+void
+ebpf_core_close_context(_In_opt_ void* context)
+{
+    if (!context)
+        return;
+
+    ebpf_epoch_enter();
```

Comment:
Not needed as this code will not touch any epoch_freed memory.

Reply:
This is necessary as the release reference can trigger a call to ebpf_epoch_free, which must be done within an epoch.